quickwit/quickwit-indexing/src/actors/indexing_service.rs

```diff
@@ -955,17 +968,13 @@ mod tests {
             })
             .await
             .unwrap();
-        assert_eq!(
-            indexing_server_handle.observe().await.num_running_pipelines,
-            0
-        );
-        assert_eq!(
-            indexing_server_handle
-                .observe()
-                .await
-                .num_running_merge_pipelines,
-            0
-        );
+
+        // Let the service cleanup the merge pipelines.
+        universe.simulate_time_shift(HEARTBEAT).await;
+
+        let observation = indexing_server_handle.process_pending_and_observe().await;
+        assert_eq!(observation.num_running_pipelines, 0);
+        assert_eq!(observation.num_running_merge_pipelines, 0);
```

Comment:
Interestingly, the only reason why this unit test passed was that we never increment the `running_merge_pipelines` counter, instead, it only gets set when the actor runs through and cleans up any unused mergers. This meant that when it observed the service it was always `0` and therefore passed.